### PR TITLE
Issue/sliderview

### PR DESCRIPTION
**Pull request by xuhaibahmad**
image

### CODE REVIEW

File: common/src/main/java/co/sodalabs/view/ContextExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package co.sodalabs.view
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.os.Build
+import android.support.v4.content.ContextCompat
+import android.support.v4.graphics.drawable.DrawableCompat
+
+fun Context.getBitmapFromVectorDrawable(drawableId: Int): Bitmap {
```
Comment (kevinmmarlow) [inline common/src/main/java/co/sodalabs/view/ContextExtensions.kt]: Wait, doesn't Android auto convert to PNGs for you?

File: styled-slider-view/factory.gradle [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Instead of renaming the artifact, we copy it and rename the copy.
+android.libraryVariants.all { variant ->
```
Comment (kevinmmarlow) [inline styled-slider-view/factory.gradle]: Do we need separate scripts for every module?

File: styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * style attribute by default and this enable us the hot-theme function.
+ */
+class StyledCompatSliderView : AppCompatSeekBar {
+
+    companion object {
+        private const val MINIMUM_PROGRESS = 5
+    }
+
+    private val paint = Paint()
+    private val thumbBitmap by lazy { context.getBitmapFromVectorDrawable(R.drawable.ic_slider_thumb) }
```
Comment (kevinmmarlow) [inline styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt]: If we open source this, wouldn't you want the user to be able to set this via `styleable`?

File: styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, android.R.attr.seekBarStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        initView()
+        initProperties(attrs)
+    }
+
+    private fun initView() {
+        progressDrawable = ContextCompat.getDrawable(context, R.drawable.slider_progress)
+        thumb = null
+        paint.color = Color.parseColor("#ccffffff")
```
Comment (kevinmmarlow) [inline styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt]: Wouldn't we want this exposed via styleable too?

File: styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    private fun initProperties(attrs: AttributeSet?) {
+        val attributes = context.theme.obtainStyledAttributes(attrs, R.styleable.StyledSlider, 0, 0)
+        for (i in 0 until attributes.indexCount) {
+            when (attributes.getIndex(i)) {
+                R.styleable.StyledSlider_android_progressDrawable -> progressDrawable = attributes.getDrawable(R.styleable.StyledSlider_android_progressDrawable)
+            }
+        }
+    }
+
+    override fun onDraw(c: Canvas) {
```
Comment (kevinmmarlow) [inline styled-slider-view/src/main/java/co/sodalabs/view/StyledCompatSliderView.kt]: Also wondering about how the assets would render if the drawable for the thumb was larger than the size of the progressDrawable? Do we autoscale these?

File: common/src/main/java/co/sodalabs/view/ContextExtensions.kt (hunk 1 above)
Comment (xuhaibahmad) [reply common/src/main/java/co/sodalabs/view/ContextExtensions.kt]: Nope. Not vector drawables.

File: styled-slider-view/factory.gradle (hunk 2 above)
Comment (xuhaibahmad) [reply styled-slider-view/factory.gradle]: I think @boyw165 can answer this better. I think yes, this can move to project level.

**Pull request merged**